static/app/views/issueDetails/traceTimeline/traceIssue.tsx

```diff
@@ -0,0 +1,127 @@
+  });
+  const project = projects.find(p => p.slug === event.project);
+  const issueId = event['issue.id'];
+  // We are view issue X and loading data for issue Y. This means that this call will not be cached,
+  // thus, it is a little bit slow to render the component.
+  // XXX: Create new endpoint that only fetches the metadata for the group
+  const {data: groupData, isLoading: isLoadingGroupData} = useApiQuery<Group>(
+    [
+      `/organizations/${organization.slug}/issues/${issueId}/`,
+      {query: getGroupDetailsQueryData({})},
+    ],
+    {
+      staleTime: 30000,
+      cacheTime: 30000,
+      retry: false,
+    }
+  );
+
+  // If any of data fails to load, we don't want to render the component
+  // Only "One other issue appears in the same trace. View Full Trace (X issues)" would show up
+  return (
+    <Fragment>
+      {isLoadingGroupData ? (
+        <LoadingIndicator mini />
+      ) : (
+        groupData && (
+          <TraceIssueLinkContainer
+            to={{
+              pathname: `/organizations/${organization.slug}/issues/${issueId}/events/${event.id}/`,
+              query: {
+                referrer: 'issues_trace_issue',
+              },
+            }}
+            onClick={() => {
+              trackAnalytics('issue_details.issue_tab.trace_issue_clicked', {
+                organization,
+                event_id: event.id,
+                group_id: issueId,
+              });
+            }}
+          >
+            {project && (
+              <ProjectBadge
+                project={project}
+                avatarSize={parseInt(space(4), 10)}
+                hideName
+                disableLink
+              />
+            )}
+            <TraceIssueDetailsContainer>
+              <NoOverflowDiv>
+                <TraceIssueEventTitle>
+                  {groupData.metadata.title || groupData.metadata.type}
+                </TraceIssueEventTitle>
+                <TraceIssueEventTransaction>
+                  {event.transaction}
+                </TraceIssueEventTransaction>
+              </NoOverflowDiv>
+              <NoOverflowDiv>{groupData.metadata.value}</NoOverflowDiv>
+            </TraceIssueDetailsContainer>
+          </TraceIssueLinkContainer>
+        )
+      )}
+    </Fragment>
+  );
+}
+
+const TraceIssueLinkContainer = styled(Link)`
+  display: flex;
+  gap: ${space(2)};
+  color: ${p => p.theme.textColor};
+  padding: ${space(1)} ${space(2)} ${space(1)} ${space(2)};
+  border: 1px solid ${p => p.theme.border};
+  border-radius: ${p => p.theme.borderRadius};
+  font-size: ${p => p.theme.fontSizeMedium};
+
+  &:hover {
+    background-color: ${p => p.theme.surface200};
+    color: ${p => p.theme.textColor};
+  }
+`;
+
+// The padding-right prevents overflowing the container
+// This can be changed with more CSS expertise
+const TraceIssueDetailsContainer = styled('div')`
+  padding-right: ${space(4)};
+`;
+
+const NoOverflowDiv = styled('div')`
+  white-space: nowrap;
```

Comment:
We have a util you can replace these styles with: `${p => p.theme.overflowEllipsis};`